app/src/androidTest/java/com/github/sdpcoachme/EditProfileActivityTest.kt

```diff
@@ -0,0 +1,90 @@
+        composeTestRule.onNodeWithTag("email row").assertExists("Email row elem does not exist")
+        composeTestRule.onNodeWithTag("first name row").assertExists("First name row elem does not exist")
+        composeTestRule.onNodeWithTag("last name row").assertExists("Last name row elem does not exist")
+        composeTestRule.onNodeWithTag("fav sport row").assertExists("Fav sport row elem does not exist")
+
+        //content of title row
+        composeTestRule.onNodeWithText("My Profile").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("profile pic").assertIsDisplayed()
+
+        //content of email row
+        composeTestRule.onNodeWithText("Email: ").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("email address").assertIsDisplayed()
+
+        //content of first name row
+        composeTestRule.onNodeWithText("First name: ").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("saved first name").assertIsDisplayed()
+
+        //content of last name row
+        composeTestRule.onNodeWithText("Last name: ").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("saved last name").assertIsDisplayed()
+
+        //content of sport row
+        composeTestRule.onNodeWithText("Favorite sport: ").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("saved fav sport").assertIsDisplayed()
+    }
+
+    @Test
+    fun editButtonClickActivatesCorrectElements() {
+        composeTestRule.onNodeWithTag("edit button")
+            .assertIsDisplayed()
+            .performClick()
+
+        composeTestRule.onNodeWithTag("editable first name").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("editable last name").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("editable fav sport").assertIsDisplayed()
+    }
+
+    @Test
+    fun saveButtonReturnsToCorrectElements() {
+        //change to edit mode
+        composeTestRule.onNodeWithTag("edit button")
+            .assertIsDisplayed()
+            .performClick()
+
+        //edit text fields
+        composeTestRule.onNodeWithTag("editable first name")
+            .performTextClearance()
+        composeTestRule.onNodeWithTag("editable first name")
+            .performTextInput("Updated first name")
+        Espresso.closeSoftKeyboard()
+
+        composeTestRule.onNodeWithTag("editable last name")
+            .performTextClearance()
+        composeTestRule.onNodeWithTag("editable last name")
+            .performTextInput("Updated last name")
+        Espresso.closeSoftKeyboard()
+
+        composeTestRule.onNodeWithTag("editable fav sport")
+            .performTextClearance()
+        composeTestRule.onNodeWithTag("editable fav sport")
+            .performTextInput("Updated fav sport")
+        Espresso.closeSoftKeyboard()
+
+        //save updated profile
+        composeTestRule.onNodeWithTag("save button")
+            .assertIsDisplayed()
+            .performClick()
+
+        correctInitialScreenContent()
```

Comment:
It might make sense to test that clicking the save button actually changes the text displayed for each field, that is that each field contains "Update first name", "Updated last name", ... respectively.